server/src/main/java/org/elasticsearch/action/support/TransportAction.java

```diff
@@ -18,24 +18,25 @@
  */
 
 package org.elasticsearch.action.support;
-
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.message.ParameterizedMessage;
 import org.apache.logging.log4j.Logger;
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.action.ActionRequest;
 import org.elasticsearch.action.ActionRequestValidationException;
 import org.elasticsearch.action.ActionResponse;
-import org.elasticsearch.common.component.AbstractComponent;
 import org.elasticsearch.tasks.Task;
 import org.elasticsearch.tasks.TaskListener;
 import org.elasticsearch.tasks.TaskManager;
 
 import java.util.concurrent.atomic.AtomicInteger;
 
-public abstract class TransportAction<Request extends ActionRequest, Response extends ActionResponse> extends AbstractComponent {
+public abstract class TransportAction<Request extends ActionRequest, Response extends ActionResponse> {
 
     protected final String actionName;
     private final ActionFilter[] filters;
     protected final TaskManager taskManager;
+    private static final Logger logger = LogManager.getLogger(TransportAction.class);
```

Comment:
I'm not sure that we want every action extending this to use the same logger. Also the visivility won't work, this leaves a lot of compile errors for me in "server".

Reply:
I think we should do the same thing here as we do for `BaseRestHandler`.